### PR TITLE
Handle arrays in `level` options

### PR DESCRIPTION
**Pull request by opatiny**
- feat: expose image.minMax
- feat(level): parameters can now be arrays
- test: enhance level coverage
- feat: implement enhanceContrast
- feat(drawRectangle): add fillColor = 'none' option

### CODE REVIEW

File: src/filters/enhanceContrast.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+/**
+ * Enhance the contrast of an image by spanning each channel on the range [0, image.maxValue].
+ *
+ * @param image - The image to enhance.
+ * @param options - Enhance contrast options.
+ * @returns The enhanced image.
+ */
+export function enhanceContrast(
```
Comment (targos) [inline src/filters/enhanceContrast.ts:20]: I'm not sure about the name. This is one way of enhancing contrast, but there are others and this one doesn't deserve such a generic name more, does it?

File: src/filters/level.ts [hunk 2]
```diff
@@ -94,3 +108,25 @@ export function level(image: Image, options: LevelOptions = {}) {
...
+ */
+function getValueArray(
+  value: number | number[],
+  imageChannels: number,
+): number[] {
+  if (Array.isArray(value)) {
+    if (value.length === imageChannels) {
+      return value;
+    } else {
+      throw new Error('array length is not compatible with channel option');
```
Comment (targos) [inline src/filters/level.ts:127]:
```suggestion
      throw new RangeError('array length is not compatible with channel option');
```

File: src/filters/enhanceContrast.ts (hunk 1 above)
Comment (opatiny) [reply src/filters/enhanceContrast.ts:20]: Indeed, do you have any other idea?

Comment (opatiny) [reply src/filters/enhanceContrast.ts:20]: Proposition: `spanChannelToFullRange()`

Comment (targos) [reply src/filters/enhanceContrast.ts:20]: what about `autoLevel` ?

Comment (opatiny) [reply src/filters/enhanceContrast.ts:20]: sgtm!

File: src/filters/autoLevel.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    .fill(0)
+    .map((value, index) => index);
+
+  if (image.colorModel === ImageColorModel.GREYA) {
+    channels = [0];
+  } else if (image.colorModel === ImageColorModel.RGBA) {
+    channels = [0, 1, 2];
+  }
+
+  return newImage.level({
```
Comment (targos) [inline src/filters/autoLevel.ts:39]: can't we use `image.level` directly without creating a `newImage` in this function?

**Review by targos — APPROVED** [review]

**Pull request merged**